server.go

```diff
@@ -883,6 +901,9 @@ func (s *Server) handleStream(t transport.ServerTransport, stream *transport.Str
 	method := sm[pos+1:]
 	srv, ok := s.m[service]
 	if !ok {
+		if handled := s.processUnknownStreamHandler(t, stream, trInfo); handled {
```

Comment:
We don't actually need the `dummyService` if it only contains the `nil`.

How about passing in a `nil`, and in `processStreamingRPC`, add a check on `srv`.